src/components/Checklist.jsx

```diff
@@ -0,0 +1,55 @@
+import React from "react"
+import { useState } from "react"
+import produce from "immer"
+
+const CHECKLIST_CHECKED = true
+const CHECKLIST_EMPTY = false
+
+export default function Checklist({ id, items }) {
+  const [checklist, setChecklist] = useState(
+    items.map((item) => {
+      return Array(item.items.length).fill(CHECKLIST_EMPTY)
+    })
+  )
+
+  const toggleCheck = (parentIndex, index) => {
+    setChecklist(produce(checklist, newState => {
+      newState[parentIndex][index] = !checklist[parentIndex][index]
+    }))
+  }
+
+  const renderChecklistItem = (item, index) => {
+    const titleId = `${id}-item-${index}-title`
+    const containerId = `${id}-item-${index}-container`
+    return (
+      <article key={`checklist_item_${index}`} className="checklist-item">
+        <header className="checklist-item-header">
+          <strong id={titleId} className="checklist-item-title">{item.title}</strong>
+        </header>
+        <ul id={containerId} className="checklist-sub-item-list">
+          {item.items.map((subItem, subItemIndex) => renderChecklistSubItem(subItem, subItemIndex, index))}
+        </ul>
+      </article>
+    )
+  }
+
+  const renderChecklistSubItem = (item, index, parentIndex) => {
```

Comment:
Don't need curly bracket or return statement for arrow functions if you don't want
```suggestion
  const renderChecklistSubItem = (item, index, parentIndex) => (<li key={`checklist_item_${parentIndex}_subitem_${index}`} className="checklist-sub-item">
```

Reply:
Good call, changed locally